src/plugins/intel_gpu/src/graph/layout_optimizer.cpp

```diff
@@ -1373,11 +1373,41 @@ impl_types layout_optimizer::get_forced_impl_type_by_config(program_node& node)
 #ifdef GPU_DEBUG_CONFIG
     GPU_DEBUG_GET_INSTANCE(debug_config);
     GPU_DEBUG_IF(!debug_config->forced_impl_type.empty()) {
+        // Forcing impl type of one primitive
+        std::string forced_impl_type = debug_config->forced_impl_type;
         if (node.is_type<fully_connected>()) {
-            if (debug_config->forced_impl_type == "fc:ocl")
+            if (forced_impl_type == "fc:ocl")
                 return impl_types::ocl;
-            else if (debug_config->forced_impl_type == "fc:onednn")
+            else if (forced_impl_type == "fc:onednn")
                 return impl_types::onednn;
+        } else if (node.is_type<detection_output>()) {
+            if (forced_impl_type == "do:cpu")
+                return impl_types::cpu;
+            else if (forced_impl_type == "do:ocl")
+                return impl_types::ocl;
+        } else if (node.is_type<reduce>()) {
+            if (forced_impl_type == "reduce:ocl")
+                return impl_types::ocl;
+            else if (forced_impl_type == "reduce:onednn")
+                return impl_types::onednn;
+        }
+
+        // Forcing one layer
+        size_t found_type = forced_impl_type.rfind(":");
+        if (found_type != std::string::npos) {
+            impl_types preferred_type = impl_types::any;
+            auto impl_type = forced_impl_type.substr(found_type + 1);
+            if (impl_type == "ocl")
+                preferred_type = impl_types::ocl;
+            else if (impl_type == "onednn")
+                preferred_type = impl_types::onednn;
+            else if (impl_type == "cpu")
+                preferred_type = impl_types::cpu;
+
+            if (node.id() == forced_impl_type.substr(0, found_type)) {
+                std::cout << "  >>> " << forced_impl_type.substr(0, found_type) << " : " << forced_impl_type.substr(found_type + 1) << std::endl;
+                return preferred_type;
+            }
```

Comment:
@yeonbok  is this what you want?
I think this feature needs more plan and review. What about separating this into separate PR?
At this moment, I don't think this would be useful :(

Reply:
Of course, It is totally okay to separate PR.

Reply:
Actually, my question is "is this what you want?"
As it is selecting among (cpu, ocl, onednn), I'm afraid that it may not fulfill your needs.

Reply:
I thought it meant cpu or clDNN result when it had selected  clDNN or oneDNN.
If you both prefer separated PR or more review about this feature. I will remove this feature from this PR